### ISSUE TITLE
Nightly clang+cuda build errors: KokkosSparse_spmv_bsrmatrix_impl.hpp:367: error: reference to __device__ function 'operator=' in __host__ __device__ function

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#1209]
PR #1197 included updates to address unused variables caught when compiling with -Werror. The merge of that PR exposed another preexisting issue in clang/8+cuda/10 builds within `KokkosSparse_spmv_bsrmatrix_impl`

**Error message**:
```
05:16:38 /home/jenkins/jenkins-new/workspace/KokkosKernels_KokkosDev2_CLANG8_CUDA10_cpp14/kokkos-kernels/src/sparse/impl/KokkosSparse_spmv_bsrmatrix_impl.hpp:367:46: error: reference to __device__ function 'operator=' in __host__ __device__ function
05:16:38             sa(ti / FRAG_M, ti % FRAG_M, tj) = 0;
```

**Reproducer instructions** (kokkos-dev-2 testbed):
```
source /projects/sems/modulefiles/utils/sems-modules-init.sh ; module use /home/projects/x86-64/modulefiles/local
module load sems-env sems-cmake/3.17.1 clang/8.0 cuda/10.0

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=Cuda,Serial --arch=SNB,Volta70 --compiler=clang++ --cxxflags="-O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pass-failed " --cxxstandard="14" --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft --with-options=disable_deprecated_code --with-cuda-options=enable_lambda   --no-examples
```

**Comment by ndellingwood** [comment]
Compilation errors resolved with #1307

**Comment by cwpearson** [comment]
Fixed in #1307
